irksome/galerkin_stepper.py

```diff
@@ -34,7 +34,60 @@ def getElements(basis_type, order):
     return trial_el, test_el
 
 
-def getFormGalerkin(F, L_trial, L_test, Q, t, dt, u0, stages, bcs=None):
+def getTermGalerkin(F, L_trial, L_test, Q, t, dt, u0, stages, test, aux_indices):
+    # preprocess time derivatives
+    F = expand_time_derivatives(F, t=t, timedep_coeffs=(u0,))
+    v, = F.arguments()
+    assert v.function_space() == u0.function_space()
+
+    qpts = Q.get_points()
+    qwts = Q.get_weights()
+    tabulate_trials = L_trial.tabulate(1, qpts)
+    trial_vals = tabulate_trials[(0,)]
+    trial_dvals = tabulate_trials[(1,)]
+    test_vals = L_test.tabulate(0, qpts)[(0,)]
+    test_vals_w = np.multiply(test_vals, qwts)
+
+    trial_vals = vecconst(trial_vals)
+    trial_dvals = vecconst(trial_dvals)
+    test_vals_w = vecconst(test_vals_w)
+    qpts = vecconst(np.reshape(qpts, (-1,)))
+
+    # set up the pieces we need to work with to do our substitutions
+    v_np = np.reshape(test, (-1, *u0.ufl_shape))
+    w_np = np.reshape(stages, (-1, *u0.ufl_shape))
+
+    # Expand unknowns in the trial space, including the initial condition
+    u_np = np.concatenate((np.reshape(u0, (1, *u0.ufl_shape)), w_np))
+    vsub = test_vals_w.T @ v_np
+    usub = trial_vals.T @ u_np
+    dtu0sub = trial_dvals.T @ u_np
+    if aux_indices is not None:
+        cur = 0
+        aux_comps = []
+        for field, Vsub in enumerate(v.function_space()):
+            if field in aux_indices:
+                aux_comps.extend(range(cur, cur+Vsub.value_size))
+            cur += Vsub.value_size
+
+        # Expand auxiliary variables in the test space rather than the trial space
+        test_vals = vecconst(test_vals)
+        usub[:, aux_comps] = test_vals.T @ w_np[:, aux_comps]
+
+    dtu0 = TimeDerivative(u0)
```

Comment:
I don't see why this is needed here

Reply:
Not strictly needed.  It makes the dictionary of substitutions shorter.  It's just syntactic sugar